nuxeo-services/nuxeo-platform-dublincore/src/main/java/org/nuxeo/ecm/platform/dublincore/service/DublinCoreStorageService.java

```diff
@@ -1,110 +1,37 @@
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
  *
- * Contributors:
- *     Nuxeo - initial API and implementation
- *
- * $Id: JOOoConvertPluginImpl.java 18651 2007-05-13 20:28:53Z sfermigier $
+ *  Contributors:
+ *      Nuno Cunha (ncunha@nuxeo.com)
  */
 
 package org.nuxeo.ecm.platform.dublincore.service;
 
-import static org.nuxeo.ecm.core.api.event.DocumentEventTypes.ABOUT_TO_CREATE;
-import static org.nuxeo.ecm.core.api.security.SecurityConstants.SYSTEM_USERNAME;
-
-import java.security.Principal;
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Calendar;
-import java.util.List;
 
 import org.nuxeo.ecm.core.api.DocumentModel;
-import org.nuxeo.ecm.core.api.SystemPrincipal;
 import org.nuxeo.ecm.core.event.Event;
-import org.nuxeo.ecm.core.schema.SchemaManager;
-import org.nuxeo.runtime.api.Framework;
-import org.nuxeo.runtime.model.DefaultComponent;
-
-/**
- * Service that writes Metadata.
- *
- * @author <a href="td@nuxeo.com">Thierry Delprat</a>
- */
-public class DublinCoreStorageService extends DefaultComponent {
-
-    public static final String ID = "DublinCoreStorageService";
-
-    public void setCreationDate(DocumentModel doc, Calendar creationDate, Event event) {
-        doc.setProperty("dublincore", "created", creationDate);
-        addContributor(doc, event);
-    }
-
-    public void setModificationDate(DocumentModel doc, Calendar modificationDate, Event event) {
-        doc.setProperty("dublincore", "modified", modificationDate);
-        if (doc.getProperty("dublincore", "created") == null) {
-            setCreationDate(doc, modificationDate, event);
-        }
-    }
-
-    public void addContributor(DocumentModel doc, Event event) {
-        Principal principal = event.getContext().getPrincipal();
-        if (principal == null) {
-            return;
-        }
-
-        String principalName = principal.getName();
-        if (principal instanceof SystemPrincipal) {
-            SystemPrincipal nxp = (SystemPrincipal) principal;
-            String originatingUser = nxp.getOriginatingUser();
-            if ((originatingUser == null || SYSTEM_USERNAME.equals(originatingUser))
-                    && !ABOUT_TO_CREATE.equals(event.getName())) {
-                return;
-            } else {
-                principalName = originatingUser;
-            }
-        }
-
-        String[] contributorsArray = (String[]) doc.getProperty("dublincore", "contributors");
 
-        List<String> contributorsList = new ArrayList<String>();
+public interface DublinCoreStorageService {
```

Comment:
Can you add `@since 10.2` in javadoc ? and maybe stating that formerly this was a class, wdyt ?